### DISCUSSION TITLE
What config is missing to avoid "React is not defined"?

### DISCUSSION THREAD

**Discussion by BlessYAHU** [loreanvictor/callbag-jsx#19] [Q&A]
I setup the todo example in a [codesandbox](https://codesandbox.io/s/objective-elgamal-6t64sp?file=/src/index.tsx ), and I get "React is not defined" when I use the jsx fragment "<></>".  I thought I mirrored the setup in the [stackblitz example](https://stackblitz.com/edit/callbag-jsx-todolist).  What config could I have missed to cause that error message?

**Comment by loreanvictor** [comment]
This is weird, it seems the sandbox is not picking up the typescript config. perhaps it is using a bundler that doesn't necessarily use all of the options outlined in tsconfig?

**Reply by BlessYAHU** [reply]
as I was tinkering, I changed the compiler option for jsxFactory to "renderer.fragment" and got "renderer.fragment is not a function".  I'm using callbag-jsx 1.5.1.  Code Sandbox is doing something strange not finding that method. 🤔 
![image](https://github.com/loreanvictor/callbag-jsx/assets/317033/7217f8b8-d627-401d-a029-545bb8d10c3f)

**Reply by loreanvictor** [reply]
Yeah the expected type of `jsxFactory` and `jsxFragmentFactory` are different, despite the name suggesting both should be functions. This is I think because of [how React used to handle fragments](https://legacy.reactjs.org/docs/fragments.html). In any case, `jsxFactory` is a function that is called with JSX values (the tag, the properties, the children), while `jsxFragmentFactory` is a special tag that is passed to the `jsxFactory` function to instruct it to create a fragment.

**Reply by BlessYAHU** [reply]
Thank you for explaining.

I guess we can chalk this up to code sandbox issue, because it does work in StackBlitz.
